cirq/circuits/circuit_dag.py

```diff
@@ -175,8 +175,28 @@ def get_next_node(succ: networkx.classes.coreviews.AtlasView
     def all_operations(self) -> Iterator[ops.Operation]:
         return (node.val for node in self.ordered_nodes())
 
+    def all_qubits(self):
+        return frozenset(q for node in self.nodes for q in node.val.qubits)
+
     def to_circuit(self) -> circuit.Circuit:
         return circuit.Circuit.from_ops(
                     self.all_operations(),
                     strategy=circuit.InsertStrategy.EARLIEST,
                     device=self.device)
+
+    def is_topologically_sorted(self, operations: ops.OP_TREE):
```

Comment:
Add a docstring explaining what this does (...what *does* this do? Why are the operations being passed in? Aren't DAGs topologically sorted by definition?). Give examples of passing and failing operations.

Reply:
Done.

This tests that the operations are topologically sorted according to the DAG.